pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -871,12 +872,21 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 
 		inUseErr := util.CheckDeviceInUseBySystem(deviceToDetach.DevicePath)
 		if inUseErr != nil {
-			// On failure, return error. Caller will log and retry.
-			return fmt.Errorf(
-				"OpenFile to verify unmount failed for volume %q (spec.Name: %q) with: %v",
-				deviceToDetach.VolumeName,
-				deviceToDetach.VolumeSpec.Name(),
-				inUseErr)
+			if os.IsNotExist(inUseErr) {
+				// If device path does not exist, print warning but do not fail out.
```

Comment:
@saad-ali @ciwang I know what this code is about and reading this I admit I am still confused. Maybe we can clean it up or comment it more. Or maybe it will look nicer once it is restructured with two return values. But it is tricky stuff and will probably confuse someone just reading as they debug.